src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/personalrelationships/profiledetails/ContactPersonProfileDetailsSyncService.kt

```diff
@@ -0,0 +1,37 @@
+package uk.gov.justice.digital.hmpps.prisonertonomisupdate.personalrelationships.profiledetails
+
+import com.microsoft.applicationinsights.TelemetryClient
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.prisonertonomisupdate.personalrelationships.ContactDomesticStatusCreatedEvent
+import uk.gov.justice.digital.hmpps.prisonertonomisupdate.profiledetails.ProfileDetailsNomisApiService
+
+@Service
+class ContactPersonProfileDetailsSyncService(
+  private val dpsApi: ContactPersonProfileDetailsDpsApiService,
+  private val nomisApi: ProfileDetailsNomisApiService,
+  private val telemetryClient: TelemetryClient,
+) {
+
+  suspend fun syncDomesticStatus(event: ContactDomesticStatusCreatedEvent) {
+    syncDomesticStatus(
+      prisonerNumber = event.personReference.identifiers.first { it.type == "prisonerNumber" }.value,
+      domesticStatusId = event.additionalInformation.domesticStatusId,
+    )
+  }
+
+  suspend fun syncDomesticStatus(prisonerNumber: String, domesticStatusId: Long) {
+    dpsApi.getDomesticStatus(prisonerNumber)
```

Comment:
I guess you will dealing later with these scenarios:

- the source of the update was NOMIS so should be ignored
- failures should log app insights